internal/mongo/replay_producer.go

```diff
@@ -44,7 +44,11 @@ func (r *ReplayProducer) Produce(ctx context.Context) (chan *ChangeEvent, error)
 	var events = make(chan *ChangeEvent)
 
 	go func() {
-		defer close(events)
+		defer func() {
```

Comment:
alors tu peux faire 
defer cursor.Close(ctx)
defer close(events)

car si tu as un panic dans le cursor close ca ne fera pas le close du channel

mais bon normalement tu ne devrait pas avoir de panic ;)

Reply:
Good catch, PR is up-to-date, thanks ;)

Reply:
sorry, but i am wondering if it is better to close the cursor first, and then the channel
because defer are executed in reverse, so in your case channel first and then cursor